src/daft-core/src/array/ops/hash.rs

```diff
@@ -24,72 +28,96 @@ where
     T: DaftPrimitiveType,
 {
     pub fn hash(&self, seed: Option<&UInt64Array>) -> DaftResult<UInt64Array> {
+        self.hash_with(seed, HashFunctionKind::XxHash)
+    }
+    pub fn hash_with(
+        &self,
+        seed: Option<&UInt64Array>,
+        _hash_function: HashFunctionKind,
+    ) -> DaftResult<UInt64Array> {
         let as_arrowed = self.as_arrow();
-
         let seed = seed.map(|v| v.as_arrow());
-
-        let result = kernels::hashing::hash(as_arrowed, seed)?;
-
+        let result = kernels::hashing::hash(as_arrowed, seed, _hash_function)?;
```

Comment:
Change "_hash_function" to "hash_function". We usually use the leading underscore to indicate an unused variable.